homeassistant/components/fido/sensor.py

```diff
@@ -84,45 +194,31 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     sensors = []
     for number in fido_data.client.get_phone_numbers():
         for variable in config[CONF_MONITORED_VARIABLES]:
-            sensors.append(FidoSensor(fido_data, variable, name, number))
+            sensors.append(
+                FidoSensor(fido_data, variable, name, number, SENSOR_TYPES[variable])
+            )
 
     async_add_entities(sensors, True)
 
 
 class FidoSensor(SensorEntity):
     """Implementation of a Fido sensor."""
 
-    def __init__(self, fido_data, sensor_type, name, number):
+    def __init__(self, fido_data, sensor_type, name, number, entity_description):
         """Initialize the sensor."""
+        self.entity_description = entity_description
         self.client_name = name
         self._number = number
         self.type = sensor_type
```

Comment:
`self.type` should be replaced by `self.entity_description.key`.

Reply:
No, it can be removed and used directly everywhere.

Reply:
That's what I meant. Could probably have been a bit clear 😅